repos/vstest.proj

```diff
@@ -3,10 +3,13 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
 
   <PropertyGroup>
+    <OverrideTargetRid>$(TargetRid)</OverrideTargetRid>
+    <OverrideTargetRid Condition="'$(TargetOS)' == 'FreeBSD'">freebsd-x64</OverrideTargetRid>
```

Comment:
Sorry if this is already obvious to everyone else: but what is the original value of `TargetRid` here that's being overridden?

Reply:
We generate this at the very beginning of the build here: https://github.com/dotnet/source-build/blob/8ebb1219998b0c360ec9ba3381c9c01255d0a1ef/tools-local/init-build.proj#L111
That calls into [our task](https://github.com/dotnet/source-build/blob/8ebb1219998b0c360ec9ba3381c9c01255d0a1ef/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/GetHostInformation.cs) which in turn is calling into [core-setup code](https://github.com/dotnet/source-build/blob/8ebb1219998b0c360ec9ba3381c9c01255d0a1ef/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj#L16) to get the RID the same way the product does.  This will by default be a non-portable RID but it looks like FreeBSD, like Windows and OSX, always uses a portable RID - @wfurt will have to be the one to explain if you're interested in why that is.

Reply:
Oh, so this is just forcing a portable RID instead of the non-portable RID? That answers my question.

Reply:
Yes, that's correct.

Reply:
yes, we decided that last year but that change never made it to all repose. Also repos now are inconsistent so source-build is the glue to force same and expect RID. 
One day we should make it simpler but for now we do want portable RID as all versions are binary and API compatible. (at least for parts we need)